webapp/src/components/chat/chat-list/ChatList.tsx

```diff
@@ -133,18 +133,25 @@ export const ChatList: FC = () => {
 
     useEffect(() => {
         // Ensure local component state is in line with app state.
+        const nonHiddenConversations: Conversations = {};
+        for (const key in conversations) {
+            if (!conversations[key].hidden) {
+                nonHiddenConversations[key] = conversations[key];
+            }
+        }
+
         if (filterText !== '') {
             // Reapply search string to the updated conversations list.
             const filteredConversations: Conversations = {};
-            for (const key in conversations) {
-                if (conversations[key].title.toLowerCase().includes(filterText.toLowerCase())) {
-                    filteredConversations[key] = conversations[key];
+            for (const key in nonHiddenConversations) {
+                if (nonHiddenConversations[key].title.toLowerCase().includes(filterText.toLowerCase())) {
+                    filteredConversations[key] = nonHiddenConversations[key];
                 }
             }
             setConversationsView({ filteredConversations: filteredConversations });
         } else {
             // If no search string, show full conversations list.
-            setConversationsView(sortConversations(conversations));
+            setConversationsView(sortConversations(nonHiddenConversations));
         }
```

Comment:
we can cut down on the second loop by combining the conditions:

```suggestion
        const filteredConversations: Conversations = {};
        for (const key in conversations) {
            const conversation = conversations[key];
            if (
                !conversation.hidden &&
                (!filterText || conversation.title.toLowerCase().includes(filterText.toLowerCase()))
            ) {
                filteredConversations[key] = conversation;
            }
        }

        setConversationsView(filterText ? { filteredConversations } : sortConversations(filteredConversations));
```